x/wasm/internal/types/tx.proto

```diff
@@ -0,0 +1,126 @@
+  string builder = 4;
+  // InstantiatePermission access control to apply on contract creation, optional
+  AccessConfig instantiate_permission = 5;
+}
+// MsgStoreCodeResponse returns store result data.
+message MsgStoreCodeResponse{
+  // CodeID is the reference to the stored WASM code
+  string code_id = 1  [(gogoproto.customname) = "CodeID"];
+}
+
+// MsgInstantiateContract create a new smart contract instance for the given code id.
+message MsgInstantiateContract {
+  // Sender is the that actor that signed the messages
+  string sender = 1;
+  // Admin is an optional address that can execute migrations
+  string admin = 2;
+  // CodeID is the reference to the stored WASM code
+  uint64 code_id = 3 [(gogoproto.customname) = "CodeID"];
+  // Label is optional metadata to be stored with a contract instance.
+  string label = 4;
+  // InitMsg json encoded message to be passed to the contract on instantiation
+  bytes init_msg = 5 [(gogoproto.casttype) = "encoding/json.RawMessage"];
+  // InitFunds coins that are transferred to the contract on instantiation
+  repeated cosmos.base.v1beta1.Coin init_funds = 6 [(gogoproto.nullable) = false, (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"];
+}
+// MsgInstantiateContractResponse return instantiation result data
+message MsgInstantiateContractResponse {
+  // Address is the bech32 address of the new contract instance.
+  string address = 1;
+}
+
+// MsgExecuteContract submits the given message data to a smart contract
+message MsgExecuteContract {
+  // Sender is the that actor that signed the messages
+  string sender = 1;
+  // Contract is the address of the smart contract
+  string contract = 2;
+  // Msg json encoded message to be passed to the contract
+  bytes msg = 3 [(gogoproto.casttype) = "encoding/json.RawMessage"];
+  // SentFunds coins that are transferred to the contract on execution
+  repeated cosmos.base.v1beta1.Coin sent_funds = 5 [(gogoproto.nullable) = false, (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"];
+}
+
+// MsgExecuteContractResponse returns execution result data.
+message MsgExecuteContractResponse {
+  // Data contains base64-encoded bytes to returned from the contract
+  bytes data = 1;
+}
+
+// MsgMigrateContract runs a code upgrade/ downgrade for a smart contract
+message MsgMigrateContract {
+  // Sender is the that actor that signed the messages
+  string sender = 1;
+  // Contract is the address of the smart contract
+  string contract = 2;
+  // CodeID references the new WASM code
+  uint64 code_id = 3 [(gogoproto.customname) = "CodeID"];
+  // MigrateMsg json encoded message to be passed to the contract on migration
+  bytes migrate_msg = 4 [(gogoproto.casttype) = "encoding/json.RawMessage"];
+}
+
+// MsgMigrateContractResponse returns contract migration result data.
+message MsgMigrateContractResponse {
+  // Data contains base64-encoded bytes returned from the wasm contract.
```

Comment:
Why are they base64-encoded? We can return the raw bytes there, right? And let the proto-buf code handle base64 or other encoding

Reply:
It actually returns the same raw bytes, updating the comment